### ISSUE TITLE
School rating hierarchy doesn't get ordered with typical sorting functions

### ISSUE THREAD

**Issue by stevevance** [stevevance/cps-school-ratings#1]
The CPS school rating hierarchy has "Level 1+" being the best (1st) rating and "Level 1" as the second best (2nd) rating. However, typical sorting functions (`sort` in PHP and `unnest(array['1+', '1', '2+', '2', '3'])` in PostgreSQL will sort resulting in "Level 1" being the 1st rating. 

This screenshot shows you the rating system on a [CPS school profile page](http://cps.edu/Schools/Pages/school.aspx?SchoolID=400125).
![screenshot 2015-01-03 22 58 38](https://cloud.githubusercontent.com/assets/1330459/5604834/d9d0e856-939c-11e4-9aad-f97460b26e72.png)

**What's the best way to order a list of schools by their rating system?**

**Comment by hunterowens** [comment]
How are you serving the data?

**Comment by stevevance** [comment]
@hunterowens The data will be served from a PostgreSQL database into PHP front end. It would be easy to create a dictionary or custom sorting function within PHP but how would I sort it properly within PostgreSQL (which may be faster than sorting with PHP, and I might want to do analysis purely within the database)?

**Comment by hunterowens** [comment]
Two Options:
1) Write a wrapper function around php `sort` that handles the edge case.
2) on ETL, bump every other level down 1, plug 1+ in as 1, then write a display wrapper that makes the changes need on the frontend.

**Comment by stevevance** [comment]
There's no opportunity to do that in my ETL. I'll import a CSV file using Navicat, a database GUI app. 

How do I write a wrapper function (in either PHP or the database) that does what you said? I presume this is what you meant in (2):

```
1+ => 1
1 => 2
2+ => 3
2 => 4
3 => 5
```

Dennis G. on Twitter suggested using [ENUM type in PostgreSQL](http://www.postgresql.org/docs/9.3/static/datatype-enum.html). That looks like it'll work because it orders based on the order used in creating the ENUM field.

**Comment by hunterowens** [comment]
Ya, that could work, I think. 

Other option is store 2 rows. (ie, 1 is `enum/string` and 1 is nums with conversion.)

Does php have case statements or pattern matching?

**Comment by hunterowens** [comment]
Anyways, Good Luck!

**Comment by stevevance** [comment]
Yes, PHP has switch/case statements and pattern matching. That's the easy part in my book. I mean, I could also just make a dictionary with `$ranking = ["1+", "1", "2+", "2", "3"];` and then write a custom sort function (which I have no idea how to do). 

I'm setting up PostgreSQL with the ENUM thing now (never used ENUM before).

**Comment by mikelikesbikes** [comment]
After some googling I found this: http://stackoverflow.com/a/4088921

It seems you can just build a custom order by expression. Something like:

```
ORDER BY idx(array['1+','1','2+','2','3',''], s.rating)
```

**Comment by stevevance** [comment]
@mikelikesbikes That appears easier than having to create ENUM type columns. Thanks for the link!
